coreneuron/apps/corenrn_parameters.cpp

```diff
@@ -0,0 +1,178 @@
+
+    auto sub_spike = app.add_option_group("spike", "Spike exchange options.");
+    sub_spike -> add_option("--ms-phases", this->ms_phases, "Number of multisend phases, 1 or 2.", true)
+        ->check(CLI::Range(1, 2));
+    sub_spike -> add_option("--ms-subintervals", this->ms_subint, "Number of multisend subintervals, 1 or 2.", true)
+        ->check(CLI::Range(1, 2));
+    sub_spike -> add_flag("--multisend", this->multisend, "Use Multisend spike exchange instead of Allgather.");
+    sub_spike -> add_option("--spkcompress", this->spkcompress, "Spike compression. Up to ARG are exchanged during MPI_Allgather.", true)
+        ->check(CLI::Range(0, 100'000));
+    sub_spike->add_flag("--binqueue", this->binqueue, "Use bin queue." );
+
+    auto sub_config = app.add_option_group("config", "Config options.");
+    sub_config -> add_option("-b, --spikebuf", this->spikebuf, "Spike buffer size.", true)
+        ->check(CLI::Range(0, 2'000'000'000));
+    sub_config -> add_option("-g, --prcellgid", this->prcellgid, "Output prcellstate information for the gid NUMBER.")
+        ->check(CLI::Range(-1, 2'000'000'000));
+    sub_config -> add_option("-k, --forwardskip", this->forwardskip, "Forwardskip to TIME")
+        ->check(CLI::Range(0., 1e9));
+    sub_config -> add_option("-l, --celsius", this->celsius, "Temperature in degC. The default value is set in defaults.dat or else is 34.0.", true)
+        ->check(CLI::Range(-1000., 1000.));
+    sub_config -> add_option("-x, --extracon", this->extracon, "Number of extra random connections in each thread to other duplicate models.")
+        ->check(CLI::Range(0, 10'000'000));
+    sub_config -> add_option("-z, --multiple", this->multiple, "Model duplication factor. Model size is normal size * multiple")
+        ->check(CLI::Range(1, 10'000'000));
+    sub_config -> add_option("--mindelay", this->mindelay, "Maximum integration interval (likely reduced by minimum NetCon delay).", true)
+        ->check(CLI::Range(0., 1e9));
+    sub_config -> add_option("--report-buffer-size", this->report_buff_size, "Size in MB of the report buffer.")
+        ->check(CLI::Range(1, 128));
+
+    auto sub_output = app.add_option_group("output", "Output configuration.");
+    sub_output -> add_option("-i, --dt_io", this->dt_io, "Dt of I/O.", true)
+        ->check(CLI::Range(-1000., 1e9));
+    sub_output -> add_option("-o, --outpath", this->outpath, "Path to place output data files.", true);
+    sub_output -> add_option("--checkpoint", this->checkpointpath, "Enable checkpoint and specify directory to store related files.");
+
+    CLI::retire_option(app, "--show");
+};
+
+void corenrn_parameters::parse (int argc, char** argv) {
+
+    try {
+        app.parse(argc, argv);
+    } catch (const CLI::ExtrasError &e) {
+
+        std::cerr << "Single-dash arguments such as -mpi are deprecated, please check ./coreneuron_exec --help for more information. \n" << std::endl;
```

Comment:
```suggestion
        std::cerr << "Single-dash arguments such as -mpi are removed, please check ./coreneuron_exec --help for more information. \n" << std::endl;
```
it's not just deprecated but removed, isn't it?